File: management-domain/Finance-department/Backend/Java/budget-tracking-service/src/main/java/com/gogidix/finance/budgettracking/shared/base/BaseEntity.java
package com.gogidix.finance.budgettracking.shared.base;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.Instant;

@Data
@NoArgsConstructor
public abstract class BaseEntity {

    @Id
    protected String id;

    @CreatedDate
    protected Instant createdAt;

    @LastModifiedDate
    protected Instant updatedAt;

    @org.springframework.data.annotation.LastModifiedBy
    @CreatedBy
    protected String updatedBy;
}

File: management-domain/Executive-domain-X/Executive-domain-X/Backend/Nodes/websocket-service/src/shared/base/base.entity.ts
import { v4 as uuidv4 } from 'uuid';
export abstract class BaseEntity { readonly id: string; readonly createdAt: Date; readonly updatedAt: Date; constructor(p?: { id?: string; createdAt?: Date; updatedAt?: Date }) { this.id = p?.id || uuidv4(); this.createdAt = p?.createdAt || new Date(); this.updatedAt = p?.updatedAt || new Date(); } }
